source/loader/layers/sanitizer/ur_sanddi.cpp

```diff
@@ -73,6 +73,12 @@ __urdlllocal ur_result_t UR_APICALL urAdapterGet(
         return UR_RESULT_ERROR_UNSUPPORTED_FEATURE;
     }
 
+    // Set env AllocateHostAllocationsInHeapExtendedHost to 0, then host and
+    // shared usm will have same allocation range on GPU device. This will
+    // simplify shadow memory mapping logic on GPU device.
+    setenv("NEOReadDebugKeys", "1", 1);
```

Comment:
@nrspruit is it a good idea to set these env variables in the layer? If this is really required, wouldn't it be a better idea to make this into an L0 API?
Using a debug key for functionality doesn't seem like a good idea.

Reply:
I don't feel comfortable merging this patch. We need confirmation from the NEO folks that this makes sense.
@MichalMrozek can you help?

Reply:
No, this is not correct to set this.
Please do not merge it.
If you want to have specific release keys avaialable this needs discussion with Piotr Rozenfeld.

Reply:
AllocateHostAllocationsInHeapExtendedHost is a debug key, debug keys cannot be used in production software. They are meant only for experiments. Driver team may remove or change this debug variable at any point of time without any notifications.

Reply:
Ok, I will send this request to driver team. Thanks.